Python/basicrequest.py

```diff
@@ -179,10 +181,13 @@ def awaitResponse(msgId, correlId):
 
         except pymqi.MQMIError as e:
             if e.comp == pymqi.CMQC.MQCC_FAILED and e.reason == pymqi.CMQC.MQRC_NO_MSG_AVAILABLE:
+                print("QM is empty")
                 # No messages, that's OK, we can ignore it.
                 pass
             else:
                 # Some other error condition.
+                qmgr.backout()
```

Comment:
you are rolling back the reply if there is an error, but there is no corresponding commit if there is no error. The reply is for this app on this reply queue. Although if there is an error, then there was an error fetching the reply, so not sure what you are backing out.